python/ray/autoscaler/_private/autoscaler.py

```diff
@@ -407,6 +415,18 @@ def _update(self):
             )
             return
 
+        # Populate mapping of replica IDs to nodes in that replica.
+        self.replicas_to_nodes.clear()
+        for node_id in self.non_terminated_nodes.worker_ids:
+            tags = self.provider.node_tags(node_id)
+            if TAG_RAY_REPLICA_INDEX in tags:
+                node_replica_index = tags[TAG_RAY_REPLICA_INDEX]
+                if node_replica_index in self.replicas_to_nodes:
+                    if node_id not in self.replicas_to_nodes[node_replica_index]:
+                        self.replicas_to_nodes[node_replica_index].append(node_id)
+                else:
+                    self.replicas_to_nodes[node_replica_index] = list(node_id)
```

Comment:
I doubted whether `list(node_id)` works or not. `node_id` seems to be a string. If we call `list("abc")`, we will get `['a','b','c']`.

Reply:
`self.replicas_to_nodes[node_replica_index].append(node_id)`

Reply:
Done in [fa083aa](https://github.com/ray-project/ray/pull/43470/commits/fa083aaffd8d110005746cce268144c92721f272).